### DISCUSSION TITLE
bitwise & operand with mask of  variables

### DISCUSSION THREAD

**Discussion by fpgiovanini** [arendst/Tasmota#22011] [General]
What is the tasmota rule syntax to perform a bitwise & operator of a variable.
EXAMPLE: %VAR%=%VAR% & 0b00001111 
tks.

**Comment by barbudor** [comment]
That simply doesn't exist
You will have to find another way

If using an ESP32, use Berry language
If using an ESP8266, use Scripting-Language instead on Rules which seems to support bit operations

**Reply by fpgiovanini** [reply]
Hi barbudor 
Thanks a lot for your reply.
best regards !!!

**Comment by fpgiovanini** [comment]
Dear Mr Barbudor
Is it possible to use RfReceived data using Scripting-Language?

**Reply by barbudor** [reply]
Probably but I have no experience with Scripting-Language

**Comment by fpgiovanini** [comment]
Thanks Mr.Barbudor but I didnt find anything till now.
Best regards

**Reply by sfromis** [reply]
I'd expect processing of `RfReceived` to be like picking data from other payloads, like sensors. Example of such:
https://tasmota.github.io/docs/Scripting-Language/#sensor-logging
But you really have to take the steep learning curve of getting started with scripting.

**Reply by pkkrusty** [reply]
Yeah, sfromis is right. All that can be done in scripting but will take some time to learn it.
